plugins/dubbo/src/main/java/com/megaease/easeagent/plugin/dubbo/AlibabaDubboCtxUtils.java

```diff
@@ -0,0 +1,233 @@
+
+			AlibabaDubboClientRequest alibabaDubboClientRequest = new AlibabaDubboClientRequest(invoker, invocation);
+			RequestContext requestContext = context.clientRequest(alibabaDubboClientRequest);
+			Span span = requestContext.span().start();
+			span.kind(alibabaDubboClientRequest.kind());
+			span.name(alibabaDubboClientRequest.name());
+			span.remoteServiceName(ConfigConst.Namespace.DUBBO);
+			span.remoteIpAndPort(rpcContext.getRemoteHost(), rpcContext.getRemotePort());
+
+			String argsList = config.argsCollectEnabled() ? JsonUtil.toJson(invocation.getArguments()) : null;
+			span.tag(DubboTags.CLIENT_APPLICATION.name, applicationName);
+			span.tag(DubboTags.GROUP.name, groupName);
+			span.tag(DubboTags.SERVICE.name, requestUrl.getPath());
+			span.tag(DubboTags.METHOD.name, method(invocation));
+			span.tag(DubboTags.SERVICE_VERSION.name, version);
+			span.tag(DubboTags.ARGS.name, argsList);
+
+			context.put(CLIENT_REQUEST_CONTEXT, requestContext);
+		} else {
+			AlibabaDubboServerRequest alibabaDubboServerRequest = new AlibabaDubboServerRequest(invoker, invocation);
+			RequestContext requestContext = context.serverReceive(alibabaDubboServerRequest);
+
+			Span span = requestContext.span().start();
+			span.kind(alibabaDubboServerRequest.kind());
+			span.name(alibabaDubboServerRequest.name());
+			span.remoteServiceName(ConfigConst.Namespace.DUBBO);
+			span.remoteIpAndPort(rpcContext.getRemoteHost(), rpcContext.getRemotePort());
+			span.tag(DubboTags.SERVER_APPLICATION.name, applicationName);
+
+			context.put(SERVICE_REQUEST_CONTEXT, requestContext);
+		}
+	}
+
+	public static void finishSpan(Context context, MethodInfo methodInfo, DubboTraceConfig config) {
+		Result result = (Result) methodInfo.getRetValue();
+		Invoker<?> invoker = (Invoker<?>) methodInfo.getArgs()[0];
+		Invocation invocation = (Invocation) methodInfo.getArgs()[1];
+		Throwable throwable = methodInfo.getThrowable();
+
+		boolean isConsumer = invoker.getUrl().getParameter(SIDE_KEY, PROVIDER_SIDE).equals(CONSUMER_SIDE);
+		boolean isAsync = RpcUtils.isAsync(invoker.getUrl(), invocation);
+		if (isAsync) {
+			Future<?> f = RpcContext.getContext().getFuture();
+			if (f instanceof FutureAdapter) {
+				ResponseFuture future = ((FutureAdapter<?>) f).getFuture();
+				future.setCallback(new ResponseCallback() {
+					@Override
+					public void done(Object response) {
+						if (response == null) {
+							doFinishSpan(isConsumer, config, context, null, new IllegalStateException("invalid result value : null, expected " + Result.class.getName()));
+							return;
+						}
+						if (!(response instanceof Result)) {
+							doFinishSpan(isConsumer, config, context, null, new IllegalStateException("invalid result type :" + response.getClass() + ", expected " + Result.class.getName()));
+							return;
+						}
+						Result result = (Result) response;
+						doFinishSpan(isConsumer, config, context, result, null);
```

Comment:
when you want to pass context between threads，use
AsyncContext asyncContext = context.exportAsync();
see AsyncContext.java for specific usage